server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -803,8 +884,17 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
           var allVolumesDir = new AllVolumesDirectory(range.tableId(), value.toString());
           bw.addMutation(manager.getContext().getAmple().createDeleteMutation(allVolumesDir));
         }
+
+        lastExtent = keyExtent;
       }
 
+      // Process deletes for the highest Tablet for any files that are not part of
+      // movedMetadataEntries
+      Sets.difference(possibleMetadataDeletes, movedMetadataEntries).forEach(delete -> {
```

Comment:
What input data would cause possibleMetadataDeletes and movedMetadataEntries to not be disjoint sets?

Reply:
This is another side effect from the disjoint issue as explained in https://github.com/apache/accumulo/pull/3640#discussion_r1298545382